src/viewPanel/sessionWidget.ts

```diff
@@ -130,9 +130,12 @@ export class SessionWidget extends BoxPanel {
 
     const future = kernel.requestExecute({ code }, false);
     await future.done;
+    this._dataLoaded.resolve();
```

Comment:
The `future. done` promise does not guarantee that data is loaded. So we should be careful when using the  `this._dataLoaded` promise.

Reply:
Actually my variable name was misleading, the fact that the data is loaded or not is not so important. We just need to make sure that the `GLUE_SESSION` variable is set before creating any tab view. I will rename the promise delegate.